src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -33,6 +33,13 @@ internal class RuntimeMetrics : IDisposable
         internal static readonly AssemblyName AssemblyName = typeof(RuntimeMetrics).Assembly.GetName();
         internal static readonly string InstrumentationName = AssemblyName.Name;
         internal static readonly string InstrumentationVersion = AssemblyName.Version.ToString();
+        private static readonly string[] HeapNames = new string[] { "gen0", "gen1", "gen2", "loh", "poh" };
+        private static readonly int NumberOfGenerations = 3;
+        private static readonly Measurement<long>[] GarbageCollectionCountMeasurements = new Measurement<long>[NumberOfGenerations];
+#if NET6_0_OR_GREATER
+        private static readonly Measurement<long>[] HeapSizeMeasurements = new Measurement<long>[GC.GetGCMemoryInfo().GenerationInfo.Length];
```

Comment:
Why do we need this variable? Is it thread safe?

Reply:
Would this work better?

https://github.com/open-telemetry/opentelemetry-dotnet/blob/c07917913cbe617ae05957f7023d5574ad7bf5d5/docs/metrics/learning-more-instruments/Program.cs#L54-L60

Reply:
I've fixed it.

I've refactored the array allocations with `yield return`. It's working well for two of the functions. 

The compile will fail with GetGarbageCollectionHeapSizes because it needs a `Span<T>`: "CS4013 Instance of type 'ReadOnlySpan<GCGenerationInfo>' cannot be used inside a nested function, query expression, iterator block or async method".

For this one I just allocated a plain array.